interface/ceed-basis.c

```diff
@@ -1303,82 +1307,89 @@ int CeedSymmetricSchurDecomposition(Ceed ceed, CeedScalar *mat,
   for (CeedInt i=0; i<n-1; i++) {
     // Calculate Householder vector, magnitude
     CeedScalar sigma = 0.0;
-    v[i] = matT[i+n*(i+1)];
+    v[i] = mat_T[i+n*(i+1)];
     for (CeedInt j=i+1; j<n-1; j++) {
-      v[j] = matT[i+n*(j+1)];
+      v[j] = mat_T[i+n*(j+1)];
       sigma += v[j] * v[j];
     }
     CeedScalar norm = sqrt(v[i]*v[i] + sigma); // norm of v[i:n-1]
-    CeedScalar Rii = -copysign(norm, v[i]);
-    v[i] -= Rii;
+    CeedScalar R_ii = -copysign(norm, v[i]);
+    v[i] -= R_ii;
     // norm of v[i:m] after modification above and scaling below
     //   norm = sqrt(v[i]*v[i] + sigma) / v[i];
     //   tau = 2 / (norm*norm)
-    if (sigma > 10*CEED_EPSILON)
+    if (sigma > CEED_EPSILON)
       tau[i] = 2 * v[i]*v[i] / (v[i]*v[i] + sigma);
     else
       tau[i] = 0;
 
+    // Don't apply Householder reflector if row already diagonal
+    if (sigma < CEED_EPSILON) continue;
```

Comment:
Why not? It's harmless and consistent.

Reply:
It was adding small (~e-15/16) noise that was making the Python/C differences even larger

Reply:
Woops, no this is for fixing #645